howto/diagnose-with-tracing/open-telemetry-collector.md

```diff
@@ -0,0 +1,254 @@
+# OpenTelemetry configurations
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-collector-conf
+  labels:
+    app: opentelemetry
+    component: otel-collector-conf
+data:
+  otel-collector-config: |
+    receivers:
+      opencensus:
+        endpoint: 0.0.0.0:55678
+    processors:
+      queued_retry:
+      batch:
+    extensions:
+      health_check:
+      pprof:
+        endpoint: :1888
+      zpages:
+        endpoint: :55679
+    exporters:
+      azuremonitor:
+      azuremonitor/2:
+        # Configuring OpenTelemetry Collector to send data to your
+        # Application Insights instance
+        endpoint: "https://dc.services.visualstudio.com/v2/track"
+        instrumentation_key: "$APP_INSIGHT_KEY"
+        # maxbatchsize is the maximum number of items that can be queued
+        # before calling to the configured endpoint
+        maxbatchsize: 100
+        # maxbatchinterval is the maximum time to wait before calling
+        # the configured endpoint.
+        maxbatchinterval: 10s
+    service:
+      extensions: [pprof, zpages, health_check]
+      pipelines:
+        traces:
+          receivers: [opencensus]
+          exporters: [azuremonitor/2]
+          processors: [batch, queued_retry]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  labels:
+    app: opencesus
+    component: otel-collector
+spec:
+  ports:
+  - name: opencensus # Default endpoint for Opencensus receiver.
+    port: 55678
+    protocol: TCP
+    targetPort: 55678
+  selector:
+    component: otel-collector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-collector
+  labels:
+    app: opentelemetry
+    component: otel-collector
+spec:
+  replicas: 1  # scale out based on your usage
+  selector:
+    matchLabels:
+      app: opentelemetry
+  template:
+    metadata:
+      labels:
+        app: opentelemetry
+        component: otel-collector
+    spec:
+      containers:
+      - name: otel-collector
+        image: otel/opentelemetry-collector-contrib-dev:latest
+        command:
+          - "/otelcontribcol"
+          - "--config=/conf/otel-collector-config.yaml"
+        resources:
+          limits:
+            cpu: 1
+            memory: 2Gi
+          requests:
+            cpu: 200m
+            memory: 400Mi
+        ports:
+          - containerPort: 55678 # Default endpoint for Opencensus receiver.
+        volumeMounts:
+          - name: otel-collector-config-vol
+            mountPath: /conf
+          #- name: otel-collector-secrets
+          #  mountPath: /secrets
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133
+      volumes:
+        - configMap:
+            name: otel-collector-conf
+            items:
+              - key: otel-collector-config
+                path: otel-collector-config.yaml
+          name: otel-collector-config-vol
+EOF
```

Comment:
For consistency can you do these as apply this command to this file file rather than as single embedded "command and file. " 
For example  kubectl apply -f  opentelemetry.yaml

Reply:
I can do that to be consistent. However, that would require the users to be in a certain directory (checking out the docs in a repo) or will need to use a hyperlink. Using a hyperlink is probably more convenient.

@msfussell @orizohar @AaronCrawfis let me know which approach above is the preferred choice?

Reply:
I recommend "create a file names `opentelemetry.yaml` with this content:

```yaml
CONTENTS
```

Apply this configuration with `kubectl apply -f opentelemetry.yaml`

Reply:
@AaronCrawfis so, something like this?

1. Create a file named opentelemetry.yaml
```
cat > opentelemetery.yaml <<EOF
CONTENTS
EOF
```

2. Apply ...

Since the user will need to copy the content anyway, might as well have them copy a few more lines for convenience.

Reply:
Updated. Please take a look.